garage/logger/outputs.py

```diff
@@ -0,0 +1,178 @@
+"""Contains the output classes for the logger.
+
+Each class is sent logger data and handles it itself.
+"""
+import abc
+import csv
+import datetime
+import os
+import sys
+
+import dateutil.tz
+
+from garage.logger.tabular_input import TabularInput
+from garage.misc.console import mkdir_p
+
+
+class LogOutput(abc.ABC):
+    """Abstract class for Logger Outputs."""
+
+    @property
+    def types_accepted(self):
+        """Pass these types to this logger output.
+
+        The types in this tuple will be accepted by this output.
+
+        :return: A tuple containing all valid input types.
+        """
+        return ()
```

Comment:
unimplemented ABC methods usually `pass`. do you need this return type for a reason?

Reply:
The way I was thinking was that if it's unimplemented it returns an empty tuple. I'll change it to pass though

Reply:
Ok, it doesn't want me to `pass` because it's a getter for a property